packages/components/src/higher-order/with-focus-outside/index.js

```diff
@@ -79,7 +79,8 @@ export default createHigherOrderComponent(
 				}
 
 				this.blurCheckTimeout = setTimeout( () => {
-					if ( 'function' === typeof this.node.handleFocusOutside ) {
+					const eventTargetElIsWithinComponent = this.node.containerRef.current && this.node.containerRef.current.contains( event.target );
```

Comment:
Can we add a comment to explain the reasoning maybe? The focus being moved to another element inside the modal which striggers the "blur" event?

Reply:
It's actually the focus being moved outside the modal from an element _within_ the Modal that fires blur. Elements within the modal should be captured.

Reply:
So there's something I'm missing if this blurCheckTimeout is only called when we move outside the modal, there's no need for an extra check, we should call `handleFocusOutside` shouldn't we?

Reply:
Sorry Riad, I'm being really unclear here. 

What I mean is there is a `blur` event firing on a `button` within the Modal which causes the block of code above to run when it shouldn't.

After I can into a few issues today I'm actually starting to think this might be a problem with our code. Let me come back to you.

Reply:
Ok I've experimented more and I think perhaps our issue was due to the focus going down into a deeper DOM level. 

Using [an enhanced version of `Disabled`](https://github.com/WordPress/gutenberg/pull/17064) seems to resolve this issue when I use the callback to cancel the ability to tab into anchors.

Reply:
That still feels like a bug and your fix like the correct fix to me.

If a "blur" event is triggered but the focus stays inside the Modal (another element of the modal), the onFocusOutside shouldn't be triggered.

Reply:
@youknowriad Ok with some more digging I've narrowed down the issue to be due to `blur` events which fire when the document loses focus. This can happen when 

* switching to devtools
* switching to another application on your computer (eg: changing focus from Firefox and back to Chrome again)

When this happens you can check whether the document has focus with the apty named method `document.hasFocus()`.

I've added a test which checks whether the document still has focus before considering the blur event to be a valid case of "focus outside". If the document isn't focused we `preventDefault` to preserve the existing focused element.

Testing in Chrome and FF and seems to solve the issue.

Also wrote tests to cover this. Not sure about my process of mocking the `document.hasFocus` method so please could you validate it is kosher? Much appreciated.

Reply:
Another alternative might be using the Page Visibility API to monitor whether the page is focused and if it is, then assign a flag to block the calls to `handleFocusOutside()`.

```
// pseudo code high level example
document.addEventListener('visibilitychange', setFlagToBlockFocusOutside, false);
```